### PR TITLE
Make Reservation

### LINKED ISSUE #9: [3pts] Create Reservation Api
- Post request to create a new reservation made by a logged in user

### PR DESCRIPTION
**Pull request by d2ndjim**
- Added route for reservation creation
- Added create action to reservation controller

closes https://github.com/ichala/short-let-api/issues/9

### CODE REVIEW

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -1,4 +1,19 @@
...
+                                                   reserve_date: reservation_params[:reserve_date], status: 'Pending')
+      if @reservation.save
+        render json: { message: 'reservation created', status: :created }
+      else
+        render json: { message: 'reservation not created' }, status: :unprocessable_entity
+      end
+    else
+      render json: { error: 'Please log in to make a reservation' }, status: :unauthorized
+    end
+  end
```
Comment (ichala) [inline app/controllers/reservations_controller.rb]: - [ ] your code looks awesome and clean 💯  , but I think you forgot to test if the Hall is available for that day or not, we cannot have multiple reservations for the same date, so we need a small method that check if the hall is available for that date or no 

1. if yes we continue with ur logic 
2. if no we should return an error saying the given date is not available

**Review by ichala — CHANGES_REQUESTED** [review]
Hi @d2ndjim , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- No linters errors✅ 
- Clean Code✅ 

 ## Required Changes ♻️

- [ ] _**check the comments below**_


## Optional suggestions

- 🛑🛑🛑🛑🛑🛑 [OPTIONAL] please try to create a more descriptive PR so we can know what API path we test and what params are required 🛑🛑🛑🛑🛑🛑


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Review by aimalamiri — APPROVED** [review]
### **Hi 👋** 

### **Congratulations! 🎉**

**For me it looks great, but make sure to implement the change which is mentioned by Ali in the above.**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Review by ichala — APPROVED** [review]
**Hi 👋 @d2ndjim ** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Review by EidHachem — APPROVED** [review]
Approved ✅

**Pull request merged**